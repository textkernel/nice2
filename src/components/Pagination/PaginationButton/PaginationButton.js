import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import styles from './PaginationButton.scss';
import Button from '../../Button';

const { block } = bem({
    name: 'PaginationButton',
    classnames: styles,
    propsToMods: ['isActive'],
});

const PaginationButton = props => {
    const { children, isActive, ...rest } = props;

    return (
        <Button {...rest} context="link" {...block(props)} aria-current={isActive ? 'page' : null}>
            {children}
        </Button>
    );
};

PaginationButton.displayName = 'PaginationButton';

PaginationButton.propTypes = {
    children: PropTypes.number.isRequired,
    isActive: PropTypes.bool,
};

PaginationButton.defaultProps = {
    isActive: false,
};

export default PaginationButton;
