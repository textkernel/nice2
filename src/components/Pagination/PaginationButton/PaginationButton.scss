.PaginationButton {
    transition: none;

    + .PaginationButton {
        margin-left: var(--spacing-normal);
    }

    &:hover {
        color: var(--color-background);
        background-color: var(--color-brand);
        text-decoration: none;
    }

    &--isActive {
        color: var(--color-background);
        background-color: var(--color-accent);

        &:hover {
            color: var(--color-background);
            background-color: var(--color-accent);
            text-decoration: none;
        }
    }
}
