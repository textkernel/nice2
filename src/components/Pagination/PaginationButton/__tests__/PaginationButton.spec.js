import React from 'react';
import toJson from 'enzyme-to-json';
import PaginationButton from '../PaginationButton';

describe('<PaginationButton> that renders a page button', () => {
    it('should render a page button and test classes', () => {
        const wrapper = shallow(<PaginationButton data-page={1}>{1}</PaginationButton>);

        expect(toJson(wrapper)).toMatchSnapshot();

        wrapper.setProps({
            isActive: true
        });

        expect(wrapper.hasClass('PaginationButton--isActive')).toBe(true);
        expect(wrapper.prop('aria-current')).toBe('page');
    });
});
