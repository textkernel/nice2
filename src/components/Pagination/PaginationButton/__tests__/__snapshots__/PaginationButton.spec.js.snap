// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationButton> that renders a page button should render a page button and test classes 1`] = `
<Button
  className="PaginationButton"
  context="link"
  data-page={1}
  disabled={false}
  isBlock={false}
  isInline={false}
  size="normal"
  type="button"
>
  1
</Button>
`;
